components/brave_wallet/common/bitcoin_utils.h

```diff
@@ -0,0 +1,47 @@
+
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_COMMON_BITCOIN_UTILS_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_COMMON_BITCOIN_UTILS_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_wallet {
+
+// TODO(apaymyshev): support more
+enum BitcoinAddressType {
+  WitnessV0ScriptHash,
+  WitnessV0PubkeyHash,
+  kWitnessUnknown
+};
+
+struct DecodedBitcoinAddress {
+  DecodedBitcoinAddress();
+  ~DecodedBitcoinAddress();
+  DecodedBitcoinAddress(const DecodedBitcoinAddress& other);
+  DecodedBitcoinAddress& operator=(const DecodedBitcoinAddress& other);
+  DecodedBitcoinAddress(DecodedBitcoinAddress&& other);
+  DecodedBitcoinAddress& operator=(DecodedBitcoinAddress&& other);
+
+  BitcoinAddressType address_type;
+  std::vector<uint8_t> pubkey_hash;
+  uint32_t witness_version = 0;
```

Comment:
Shouldn't we only need one byte for the version?

Reply:
https://github.com/brave/brave-core/commit/b6128cbaba4bf9dab4160c02090e0fe7fca1cf2e